scripts/trifingerpro_post_submission.py

```diff
@@ -304,8 +306,113 @@ def reset_object(robot, trajectory_file):
         robot.frontend.wait_until_timeindex(t)
 
 
+def record_camera_observations(
+    object_type: str, num_observations: int
+) -> typing.List[tricamera.TriCameraObjectObservation]:
+    """
+    Record camera observations while the robot is not moving.
+
+    Args:
+        object_type: Which object to look for ("cube" or "cuboid").
+        num_observations: Number of observations that are recorded.
+
+    Returns:
+        The recorded observations.
+    """
+    object_models = {
+        "cube": "cube_v2",
+        "cuboid": "cuboid_2x2x8_v2",
+        # If no object is specified, simply use cube (object poses will be
+        # garbage if there is no actual object in the scene, but one can still
+        # use the images for other checks.  This is a bit hacky but easier then
+        # using a different camera driver.
+        None: "cube_v2",
+    }
+
+    camera_data = tricamera.SingleProcessData(history_size=num_observations)
+    model = object_tracker.get_model_by_name(object_models[object_type])
+    camera_driver = tricamera.TriCameraObjectTrackerDriver(
+        "camera60", "camera180", "camera300", model
+    )
+    camera_backend = tricamera.Backend(camera_driver, camera_data)
+    camera_frontend = tricamera.Frontend(camera_data)
+
+    # collect observations
+    observation_buffer: typing.List[tricamera.TriCameraObjectObservation] = []
+    t = camera_frontend.get_current_timeindex()
+    while len(observation_buffer) < num_observations:
+        obs = camera_frontend.get_observation(t)
+        observation_buffer.append(obs)
+        t += 1
+
+    camera_backend.shutdown()
+
+    return observation_buffer
+
+
+def check_camera_sharpness(
+    observations: typing.Sequence[tricamera.TriCameraObjectObservation],
+    log: logging.Logger,
+) -> bool:
+    """Check if all cameras are reasonable well in focus.
```

Comment:
I think it should be "reasonably". Just nitpicking because I don't  find anything else ;-)

Reply:
Thanks for catching.  This is somehow a typo I make quite often.